src/relations/pgbouncer_provider.py

```diff
@@ -131,6 +132,27 @@ def _on_database_requested(self, event: DatabaseRequestedEvent) -> None:
         extra_user_roles = event.extra_user_roles or ""
         rel_id = event.relation.id
 
+        dbs = self.charm.generate_relation_databases()
+        dbs[str(event.relation.id)] = {"name": database, "legacy": False}
+        roles = extra_user_roles.lower().split(",")
+        if "admin" in roles or "superuser" in roles or "createdb" in roles:
+            dbs["*"] = {"name": "*", "auth_dbname": database, "legacy": False}
+        self.charm.set_relation_databases(dbs)
+
+        pgb_dbs_hash = shake_128(
+            self.charm.peers.app_databag["pgb_dbs_config"].encode()
+        ).hexdigest(16)
+        for key, data in self.charm.peers.relation.data.items():
+            # We skip the leader so we don't have to wait on the defer
+            if (
+                key != self.charm.app
+                and key != self.charm.unit
+                and data.get("pgb_dbs", "") != pgb_dbs_hash
+            ):
+                logger.debug("Not all units have synced configuration")
+                event.defer()
+                return
+
```

Comment:
Defer until all the units are in sync (the database is added on all units). Do we want this?

Reply:
It is to avoid DB start using until all PGB restarts happened (to avoid disconnects)?

Reply:
Yes, the idea is to wait for all the units to be in sync and then update the relation data in one go. Otherwise, we either have to check when setting the read only endpoints or expect that the client will retry if the other units are still updating.